speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SQLDatabaseStorage.cpp

```diff
@@ -6,12 +6,12 @@
 #include <config.h>
 
 /// @brief reads input from input file, unless EOF has been reached
-#define READ_INPUT(p_inputFile, p_string)                        \
-    if (p_inputFile.eof())                                       \
-    {                                                            \
-        p_inputFile.close();                                     \
-        THROW_RPP_EXCEPTION("Reached end of file prematurely");  \
-    }                                                            \
+#define READ_INPUT(p_inputFile, p_string)                       \
+    if (p_inputFile.eof())                                      \
+    {                                                           \
+        p_inputFile.close();                                    \
+        THROW_RPP_EXCEPTION("Reached end of file prematurely"); \
```

Comment:
Shouldn't this return a boolean too then?

Reply:
Have moved the functionality of this macro to a method that does return a boolean success or not.
Have changed the macro to also get a parameter p_errorReturnValue,  such that methods using the macro can have early exits without continious if(!ReadInput(params)){return errorValue;} blocks. (Both int InsertInitialData and the bool Insert{Other}Data methods used it)